devicetypes/smartthings/fidure-thermostat.src/fidure-thermostat.groovy

```diff
@@ -308,6 +308,7 @@ def parseDescriptionAsMap(description) {
        encoding = raw[index..(index+1)]
        index+= 2;
        len =getDataLengthByType(encoding)
+       if (!len) continue
```

Comment:
This should fix the exception, but it's possible there is another bug in this multi-attribute handling. Ideally this DTH would be updated to use the zigbee library more. I would at least add a comment or log a message that there is unhandled data being dropped.

Reply:
This is an example of the message that causes the exception: `read attr - raw: B116010201141100293F081200292907 dni: B116 endpoint: 01 cluster: 0201 size: 20 attrId: 0011 result: success encoding: 29 value: 0729290012083f`

Reply:
Actually now that I look more closely there must be a bug in the multi attribute handling. The raw message includes two attributes: 0x0011 and 0x0012. Both are int16's with the first having a value of 0x083F and the second having a value of 0x0729.

Reply:
I think the best thing to do is switch to the zigbee library version which handles both read attributes and report attributes correctly